taskRequests/script.js

```diff
@@ -84,6 +93,173 @@ function openTaskDetails(id) {
+  group.appendChild(containerAsc);
+
+  const containerDesc = createSortContainer(
+    id + '-desc',
+    name,
+    iconPathDesc,
+    'desc',
+  );
+  group.appendChild(containerDesc);
+}
+
+function sortModalButtons() {
+  const assigneeAsc = document.getElementById(ASSIGNEE_COUNT);
+  const assigneeDesc = document.getElementById(ASSIGNEE_DESC);
+  const createTimeAsc = document.getElementById(CREATED_TIME);
+  const createTimeDesc = document.getElementById(CREATED_TIME_DESC);
+
+  const sortModalButtons = [
+    assigneeAsc,
+    assigneeDesc,
+    createTimeAsc,
+    createTimeDesc,
+  ];
+  let selectedButton = null;
+  function toggleSortModal() {
+    sortModal.classList.toggle('hidden');
+    backDrop.style.display = 'none';
+  }
+
+  function selectButton(button) {
+    if (selectedButton === button) {
+      selectedButton.classList.remove('selected');
+      selectedButton = null;
+      toggleSortModal();
+    } else {
+      if (selectedButton) {
+        selectedButton.classList.remove('selected');
+      }
+      selectedButton = button;
+      selectedButton.classList.add('selected');
+      toggleSortModal();
+    }
+  }
+
+  sortModalButtons.forEach((button) => {
+    if (button) {
+      if (!button.hasEventListener) {
+        button.hasEventListener = true;
+        button.addEventListener('click', () => {
+          selectButton(button);
+        });
+      }
+    }
+  });
+}
+
+function createSortContainer(id, name, iconPath, sortOrder) {
+  const container = document.createElement('div');
+  container.classList.add('sort-container', sortOrder);
+
+  container.id = id;
+
+  const nameSpan = document.createElement('span');
+  nameSpan.textContent = name;
+  const label = document.createElement('label');
+  label.appendChild(nameSpan);
+
+  const icon = document.createElement('img');
+  icon.src = iconPath;
+  icon.alt = name + ' ' + sortOrder + ' icon';
+  icon.classList.add('sort-icon', sortOrder);
+
+  label.appendChild(icon);
+  label.classList.add('sort-label');
+
+  container.appendChild(label);
+
+  return container;
+}
+
+function populateStatus() {
+  const statusList = [
+    { name: 'Approved', id: 'APPROVED' },
+    { name: 'Pending', id: 'PENDING' },
+    { name: 'Denied', id: 'DENIED' },
+  ];
+  const requestList = [
+    { name: 'Assignment', id: 'APPROVED' },
+    { name: 'Creation', id: 'PENDING' },
+  ];
+
+  statusList.map(({ name, id }) => addCheckbox(name, id, 'status-filter'));
+
+  requestList.map(({ name, id }) => addCheckbox(name, id, 'status-request'));
+
+  const sortByList = [
+    {
+      name: 'Assignee Count',
+      id: 'ASSIGNEE_COUNT',
+      iconPathAsc: '/taskRequests/assets/sort-up.svg',
+      iconPathDesc: '/taskRequests/assets/sort-down.svg',
+    },
+    {
+      name: 'Created Time',
+      id: 'CREATED_TIME',
+      iconPathAsc: '/taskRequests/assets/sort-up.svg',
+      iconPathDesc: '/taskRequests/assets/sort-down.svg',
+    },
+  ];
```

Comment:
Move these to constants file.

Reply:
I am trying to move but issue are arising